cmd/capibmadm/cmd/vpc/key/list.go

```diff
@@ -0,0 +1,146 @@
+/*
+Copyright 2023 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// Package key contains the commands to operate on vpc key resources.
+package key
+
+import (
+	"context"
+	"os"
+
+	"github.com/go-openapi/strfmt"
+	"github.com/spf13/cobra"
+
+	"github.com/IBM/vpc-go-sdk/vpcv1"
+
+	"sigs.k8s.io/cluster-api-provider-ibmcloud/cmd/capibmadm/options"
+	"sigs.k8s.io/cluster-api-provider-ibmcloud/cmd/capibmadm/printer"
+	"sigs.k8s.io/cluster-api-provider-ibmcloud/cmd/capibmadm/utils"
+	pagingUtil "sigs.k8s.io/cluster-api-provider-ibmcloud/pkg/cloud/services/utils"
+)
+
+var resourceGroupName string
+
+// ListCommand vpc key list command.
+func ListCommand() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "list",
+		Short: "List VPC key",
+		Example: `
+# List key in VPC
+export IBMCLOUD_API_KEY=<api-key>
+capibmadm vpc key list --region <region> --resource-group-name <resource-group-name>`,
+	}
+
+	options.AddCommonFlags(cmd)
+	options.AddVPCCommonFlags(cmd)
+
+	cmd.Flags().StringVar(&resourceGroupName, "resource-group-name", resourceGroupName, "IBM cloud resource group name")
+
+	cmd.RunE = func(cmd *cobra.Command, args []string) error {
+		if err := listKeys(cmd.Context(), resourceGroupName); err != nil {
+			return err
+		}
+		return nil
+	}
+
+	return cmd
+}
+
+func listKeys(ctx context.Context, resourceGroupName string) error {
+	v1, err := utils.CreateVPCService(options.GlobalOptions.VPCRegion)
+	if err != nil {
+		return err
+	}
+
+	accountID, err := utils.GetAccountID(ctx, utils.GetIAMAuth())
+	if err != nil {
+		return err
+	}
+
+	if resourceGroupName != "" {
+		_, err = utils.GetResourceGroupID(ctx, resourceGroupName, accountID)
```

Comment:
Why retrieved resource group id is unassigned?
It seems `vpcv1.ListKeysOptions` not takes resourceGroupID as a option, so you can skip this step.